presto-parser/src/test/java/com/facebook/presto/sql/parser/TestSqlParserErrorHandling.java

```diff
@@ -25,118 +26,118 @@
 {
     private static final SqlParser SQL_PARSER = new SqlParser();
 
-    @Test(expectedExceptions = ParsingException.class, expectedExceptionsMessageRegExp = "line 1:1: no viable alternative at input '<EOF>'")
-    public void testEmptyExpression()
+    @DataProvider(name = "expressions")
+    public Object[][] getExpressions()
+    {
+        return new Object[][] {
+                {"", "line 1:1: mismatched input '<EOF>'. Expecting: <expression>"},
+                {"1 + 1 x", "line 1:7: mismatched input 'x'. Expecting: '%', '*', '+', '-', '.', '/', 'AT', '[', '||', <expression>"},
+                };
+    }
+
+    @DataProvider(name = "statements")
+    public Object[][] getStatements()
+    {
+        return new Object[][] {
```

Comment:
Can you refactor the original test cases into the new test format as a separate commit? That will make it easy to see how the error messages are improved.

Reply:
Updated.